tests/collections/asr/test_asr_datasets.py

```diff
@@ -125,3 +155,164 @@ def test_tarred_bpe_dataset(self, test_data_dir):
+        device = 'gpu' if torch.cuda.is_available() else 'cpu'
+        texts = []
+
+        with tempfile.NamedTemporaryFile(mode='w', encoding='utf-8') as f:
+            with open(manifest_path, 'r') as m:
+                for ix, line in enumerate(m):
+                    if ix >= num_samples:
+                        break
+
+                    line = line.replace("tests/data/", "tests/.data/").replace("\n", "")
+                    f.write(f"{line}\n")
+
+                    data = json.loads(line)
+                    texts.append(data['text'])
+                    num_samples += 1
+
+            f.seek(0)
+
+            dataset = AudioToCharDALIDataset(
+                manifest_filepath=f.name,
+                device=device,
+                batch_size=batch_size,
+                labels=self.labels,
+                max_duration=16.0,
+                parser='en',
+                shuffle=False,
+            )
+
+            assert len(dataset) == 5  # num batches
+            count = 0
+            original_transcripts = []
+            for batch in dataset:
+                transcripts = batch[2]  # transcript index in DALIOutputs
+                transcripts_lengths = batch[3]  # transcript length index in DALIOutputs
+                transcripts = [
+                    decode_chars(transcript, transcripts_length, mapping=self.labels)
+                    for transcript, transcripts_length in zip(transcripts, transcripts_lengths)
+                ]
+                original_transcripts.extend(transcripts)
+                count += len(transcripts)
+            assert count == 10
+
+            # Repeat, now with shuffle enabled
+            f.seek(0)
+
+            dataset = AudioToCharDALIDataset(
+                manifest_filepath=f.name,
+                device=device,
+                batch_size=batch_size,
+                labels=self.labels,
+                max_duration=16.0,
+                parser='en',
+                shuffle=True,
+            )
+
+            assert len(dataset) == 5  # num batches
+            count = 0
+            shuffled_transcripts = []
+            for batch in dataset:
+                transcripts = batch[2]  # transcript index in DALIOutputs
+                transcripts_lengths = batch[3]  # transcript length index in DALIOutputs
+                transcripts = [
+                    decode_chars(transcript, transcripts_length, mapping=self.labels)
+                    for transcript, transcripts_length in zip(transcripts, transcripts_lengths)
+                ]
+                shuffled_transcripts.extend(transcripts)
+                count += len(transcripts)
+            assert count == 10
+
+            sample_matches = 0
+            for orig, shuffled in zip(original_transcripts, shuffled_transcripts):
+                if orig == shuffled:
+                    sample_matches += 1
+            assert sample_matches < 6  # assume after shuffling <half samples have displaced targets
```

Comment:
I guess that this assumption is not guaranteed. An idea: You could add an optional `seed` parameter to the DALI pipeline, so that you can make this test deterministic.